source/Mission.cpp

```diff
@@ -1349,8 +1436,19 @@ Mission Mission::Instantiate(const PlayerInfo &player, const shared_ptr<Ship> &b
 		return result;
 	}
 	for(const auto &it : actions)
-		result.actions[it.first] = it.second.Instantiate(subs, sourceSystem, jumps, payload);
-
+	{
+		if(it.first == COMPLETE && result.GetOutfit() && result.OutfitUnits() > 0)
```

Comment:
```suggestion
		if(it.first == COMPLETE && result.outfit && result.outfitUnits > 0)
```
wouldnt that be the same without an extra function call?

Reply:
Old encapsulation habit from work when I worked with some people that were very academic. Can easily change to reduce the public interface.

Reply:
yes, probably better as that is how we do it most of the time

Reply:
Change made on local